tests/src/lib.rs

```diff
@@ -185,13 +190,11 @@ async fn ws_more_request_than_buffer_should_not_deadlock() {
 
 	for _ in 0..6 {
 		let c = client.clone();
-		requests.push(tokio::spawn(async move {
-			let _: JsonValue = c.request("say_hello", Params::None).await.unwrap();
-		}));
+		requests.push(tokio::spawn(async move { c.request::<String, _, _>("say_hello", Params::None).await }));
```

Comment:
after the request manager limit, it will reject the request direct because once it can't allocate a request ID it would propagated the users, kind of fixes https://github.com/paritytech/jsonrpsee/issues/225.

It will increase channel capacity but be dropped when it reaches the background thread.